tests/integration/security/mtls_permissive_with_dns_cert_test.go

```diff
@@ -0,0 +1,190 @@
+	tlsContext := &envoy_api_v2_auth.DownstreamTlsContext{}
+	err := ptypes.UnmarshalAny(transportProtocolChain.GetTransportSocket().GetTypedConfig(), tlsContext)
+	if err == nil {
+		if !reflect.DeepEqual(tlsContext.GetCommonTlsContext().GetTlsCertificates()[0].GetCertificateChain().GetFilename(), "/etc/certs/custom/cert-chain.pem") {
+			return errors.New("tls cert file name is not /etc/certs/custom/cert-chain.pem")
+		}
+		if !reflect.DeepEqual(tlsContext.GetCommonTlsContext().GetTlsCertificates()[0].GetPrivateKey().GetFilename(), "/etc/certs/custom/key.pem") {
+			return errors.New("tls key file name is not /etc/certs/custom/key.pem")
+		}
+		if !reflect.DeepEqual(tlsContext.GetCommonTlsContext().GetValidationContext().GetTrustedCa().GetFilename(), "/etc/certs/custom/root-cert.pem") {
+			return errors.New("tls ca file name is not /etc/certs/custom/root-cert.pem")
+		}
+	} else {
+		return errors.New("unable to unmarshal tls context")
+	}
+	return nil
+}
+
+// TestTLSServerDNSCertWithAuthnPermissive checks when authentication policy is permissive,
+// and metadata value of TLS_SERVER_DNS_CERT is passed to Pilot, it generates expected
+// listener configuration with additional filter chain with match as transport_protocol: "tls"
+func TestTLSServerDNSCertWithAuthnPermissive(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "permmtls",
+				Inject: true,
+			})
+			policy := `
+apiVersion: authentication.istio.io/v1alpha1
+kind: Policy
+metadata:
+  name: default
+spec:
+  peers:
+    - mtls:
+        mode: PERMISSIVE
+`
+			g.ApplyConfigOrFail(t, ns, policy)
+
+			var b echo.Instance
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+				BuildOrFail(t)
+
+			t.Logf("echo boot warmed")
+			nodeID := b.WorkloadsOrFail(t)[0].Sidecar().NodeID()
+
+			// Set custom dir path for TLS_SERVER_DNS_CERT
+			echoNodeMetadata := &model.NodeMetadata{
+				TLSServerDNSCert: "/etc/certs/custom",
+			}
+
+			newDiscoveryRequest := &xdsapi.DiscoveryRequest{
+				Node: &xdscore.Node{
+					Id:       nodeID,
+					Metadata: echoNodeMetadata.ToStruct(),
+				},
+				TypeUrl: string(pilot.Listener),
+			}
+
+			// Start the xDS stream containing the listeners for this node
+			p.StartDiscoveryOrFail(t, newDiscoveryRequest)
+
+			p.WatchDiscoveryOrFail(t, time.Second*60,
+				func(resp *xdsapi.DiscoveryResponse) (b bool, e error) {
+					var errs []error
+					for _, r := range resp.Resources {
+						foo := &xdsapi.Listener{}
+						err := ptypes.UnmarshalAny(r, foo)
+						if err != nil {
+							errs = append(errs, err)
+							continue
+						}
+						// Only test inbound listeners
+						if foo.GetTrafficDirection() == xdscore.TrafficDirection_OUTBOUND {
+							continue
+						}
+
+						err = verifyListenerForTLSServerDNSCert(foo, t)
```

Comment:
Please do not check the listener config dump in the e2e test. This is not appropriate in e2e test because it couples the detail code logic to the external e2e tests, it's hard to maintain and doesn't verify the actual traffic works, in other words, it doesn't cover the e2e flow and bypass the Enovy logic.

Could you write the e2e tests to actually send some requests from a client and checks the connection works?